orchestrator/orchestrator.go

```diff
@@ -119,63 +130,82 @@ func (s *Orchestrator) execute(filter func(wf *process.Process, node *process.Pr
 	var processes []*process.Process
 	route := fmt.Sprintf("custom/%s/%s", processmodule.QuerierRoute, processmodule.QueryList)
 	if err := s.rpc.QueryJSON(route, nil, &processes); err != nil {
-		s.ErrC <- err
+		s.logger.Error(err.Error())
 		return
 	}
 	for _, wf := range processes {
 		for _, node := range s.findNodes(wf, filter) {
 			if err := s.executeNode(wf, node, exec, event, data); err != nil {
-				s.ErrC <- err
+				s.logger.With(keyvals(wf, node, exec, event, data)...).Error(err.Error())
 			}
 		}
 	}
 }
 
 func (s *Orchestrator) executeNode(wf *process.Process, n *process.Process_Node, exec *execution.Execution, event *event.Event, data *types.Struct) error {
-	logrus.WithField("module", "orchestrator").
-		WithField("node.key", n.Key).
-		WithField("type", fmt.Sprintf("%T", n)).Debug("process process")
-	if task := n.GetTask(); task != nil {
+	log := s.logger.With(keyvals(wf, n, exec, event, data)...)
+	// Process the node
+	switch x := n.Type.(type) {
+	case *process.Process_Node_Task_:
 		// This returns directly because a task cannot process its children.
 		// Children will be processed only when the execution is done and the dependencies are resolved
-		return s.processTask(n.Key, task, wf, exec, event, data)
-	} else if m := n.GetMap(); m != nil {
+		createdExecHash, err := s.processTask(n, x.Task, wf, exec, event, data)
+		if err != nil {
+			return err
+		}
+		log.With("createdExecHash", createdExecHash.String()).Info("execution created")
+		return nil // stop workflow execution
+	case *process.Process_Node_Map_:
 		var err error
-		data, err = s.processMap(n.Key, m.Outputs, wf, exec, data)
+		data, err = s.processMap(n.Key, x.Map.Outputs, wf, exec, data)
 		if err != nil {
 			return err
 		}
-	} else if filter := n.GetFilter(); filter != nil {
-		if !s.filterMatch(filter, wf, n.Key, exec, data) {
-			return nil
+		if result, err := json.Marshal(data); err == nil {
+			log = log.With("output", string(result))
+		}
+	case *process.Process_Node_Filter_:
+		if result, err := json.Marshal(x.Filter); err == nil {
+			log = log.With("filter", string(result))
+		}
+		if !s.filterMatch(x.Filter, wf, n, exec, data) {
+			log.Info("filter does not match data")
+			return nil // stop workflow execution
 		}
 	}
+
+	// Process the children
 	for _, childrenID := range wf.ChildrenKeys(n.Key) {
+		log = log.With("to", childrenID)
 		children, err := wf.FindNode(childrenID)
 		if err != nil {
 			// does not return an error to continue to process other tasks if needed
-			s.ErrC <- err
+			log.Error(err.Error())
 			continue
 		}
+		log.Info("executed process transition")
 		if err := s.executeNode(wf, children, exec, event, data); err != nil {
 			// does not return an error to continue to process other tasks if needed
-			s.ErrC <- err
+			s.logger.With(keyvals(wf, children, exec, event, data)...).Error(err.Error())
```

Comment:
why this one is not using `log` the context is the same.
```suggestion
			log.Error(err.Error())
```

Reply:
The context is not the same as before.
The previous transition is already done (the log is displayed line 186).
This new context corresponds to the next transition (execute the children of current node). So if there is an error, it will display it with info from the next node (current node = children, data = input and not transformed with the map).

Reply:
that's a bit confusing but yes, this log applies to the next transition even if the next transition will not occur